bats/tests/profile/sample.bats

```diff
@@ -32,69 +45,188 @@ start_app() {
     wait_for_container_engine
 }
 
-verify_settings() {
+verify_profiles() {
     PROFILE_TYPE=$PROFILE_LOCKED
     run profile_exists
-    "${assert}_success"
+    "${assert}_success" || return
 
     PROFILE_TYPE=$PROFILE_DEFAULTS
     run profile_exists
-    "${assert}_success"
+    "${assert}_success" || return
+}
 
+verify_settings() {
+    #settings from defaults profile
     run get_setting .containerEngine.name
-    "${assert}_output" "$PROFILE_CONTAINER_ENGINE"
+    "${assert}_output" "$PROFILE_CONTAINER_ENGINE" || return
```

Comment:
Yes, verify settings is not, but this `|| return` expressions were added because we want to make assertions of the output of `get_setting`, which is passed as an argument to `run`

Reply:
That's not how it works. Only code **inside** of `get_setting`, and code called **by** `get_setting` have `errexit` disabled. As soon as `run get_setting` returns, `errexit` is in effect again.